File: src/main/java/com/zoarial/iot/action/model/ExternalIoTAction.java
package com.zoarial.iot.action.model;

import lombok.NoArgsConstructor;

import jakarta.persistence.Entity;

@Entity
@NoArgsConstructor
public class ExternalIoTAction extends IoTAction {
    @Override
    public boolean isValid() {
        return true;
    }

    @Override
    protected String privExecute(IoTActionArgumentList args) {
        // TODO: do something similar to JavaIoTAction where a socket is queried when run
        return "This is an external action. You shouldn't be seeing this.";
    }
}

File: src/main/java/com/zoarial/iot/action/model/ExternalIoTAction.kt
package com.zoarial.iot.action.model

import jakarta.persistence.Entity
import lombok.NoArgsConstructor
import java.lang.RuntimeException

@Entity
@NoArgsConstructor
class ExternalIoTAction : IoTAction() {
    override val isValid: Boolean
        get() = true

    override fun privExecute(args: IoTActionArgumentList): String {
        // TODO: do something similar to JavaIoTAction where a socket is queried when run
        throw RuntimeException("Attempting to execute a remote action.")
    }
}